### PR TITLE
Version 1.10.0

### PR DESCRIPTION
**Pull request by bulkware**
- Improve documentation.
- Read certificate thumbprint using pyOpenSSL.
- Switch adal to msal.

### CODE REVIEW

**Review by sapajune — APPROVED** [review]
Looks good.

File: README.md [hunk 1]
```diff
@@ -47,6 +49,18 @@ azure.discover.resources[configuration_file] | Discover resources from Azure's s
...
 azure.discover.metrics[configuration_file, resource] | Discover metrics from Azure's resources | {#METRIC_CATEGORY}, {#METRIC_NAME} |
+azure.discover.metrics.namespace[configuration_file, resource, metric namespace] | Discover metrics from Azure's resources using metric namespace. | {#METRIC_CATEGORY}, {#METRIC_NAME} |
+
+
+
+### Dimension discovery
+
+Item Syntax | Description | Units |
+----------- | ----------- | ----- |
+azure.discover.dimensions[configuration_file, resource, metric_category/metric_name, dimension] | Discover dimensions from Azure's resources | {#DIMENSION} |
```
Comment (kompa3) [inline README.md:60]: I suppose that dimension discovery does not need a dimension as the argument?

**Review by kompa3 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bulkware) [reply README.md:60]: It actually does need the dimension. Dimension should be for example"cloud/roleName" and the resulting macros could be, for example the name for a dependency or such.

Comment (kompa3) [reply README.md:60]: Hmm is there a contradiction then concerning the terms here?

Comment (bulkware) [reply README.md:60]: It probably should be something like "dimension_key" and the result would be "dimension_value(s)" but never needed to change it. The documentation actually refers to those in that way, but not in the metric list which we have used previously:
https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/metrics-custom-overview?toc=%2Fazure%2Fazure-monitor%2Ftoc.json#dimension-keys

Comment (bulkware) [reply README.md:60]: @kompa3 should I change the parameter to "dimension-key" and the values that are sent back to "{#DIMENSION_VALUE}"? We could keep the old macro "{#DIMENSION}" for backwards compatibility.

**Pull request merged**